### PR TITLE
Support for parameter-dependent conservation laws

### LINKED ISSUE #1673: Support for parameter-dependent conservation laws
So far, conservation laws can only be state-dependent. It would be great if they could also be parameter-dependent to cope, for example, with parameterized compartment sizes.

### PR DESCRIPTION
**Pull request by dweindl**
Adds support for parameter-dependent conservation laws. This allows applying conservation laws to models comprising species  in different compartments where model states correspond to concentrations. 

Closes #1673

### CODE REVIEW

File: python/amici/ode_export.py [hunk 1]
```diff
@@ -1854,8 +1870,39 @@ def _compute_equation(self, name: str) -> None:
...
 
         elif name == 'dtcldp':
-            # force symbols
-            self._eqs[name] = self.sym(name)
+            self._derivative('total_cl', 'p', name=name)
+
+        elif name == 'stotal_cl':
+            # stotal_cl = dtotal_cl/dp +  dtotal_cl/dx_rdata * sx_rdata
+            # shape: ncl x np
+            dtotal_cldp = smart_jacobian(self.eq('total_cl'), self.sym('p'))
```
Comment (FFroehlich) [inline python/amici/ode_export.py:1878]:
```suggestion
            self._eqs[name] = self.eq('tcldp')
```

File: python/amici/ode_export.py [hunk 2]
```diff
@@ -1854,8 +1870,39 @@ def _compute_equation(self, name: str) -> None:
...
+            dtotal_cldx_rdata = smart_jacobian(self.eq('total_cl'),
+                                               self.sym('x_rdata'))
+            self._eqs[name] = dtotal_cldp
```
Comment (FFroehlich) [inline python/amici/ode_export.py:1881]:
```suggestion
```

File: python/amici/ode_export.py [hunk 3]
```diff
@@ -1854,8 +1870,39 @@ def _compute_equation(self, name: str) -> None:
...
+            tmp = smart_multiply(dtotal_cldx_rdata, self.sym('sx_rdata'))
+            for ip in range(self._eqs[name].shape[1]):
+                self._eqs[name][:, ip] += tmp
+        elif name == 'sx_rdata':
+            # sx_rdata =  dx_rdata/dx_solver * sx_solver
+            #              + dx_rdata/d_tcl * stcl + dxrdata/dp
+            # shape: nx_rdata, 1
+            dx_rdata_dx_solver = smart_jacobian(self.eq('x_rdata'),
```
Comment (FFroehlich) [inline python/amici/ode_export.py:1889]: if we don't have any cls, we can simply `self._eqs[name] = self.sym('sx_solver')`, will probably speed things up quite a bit for larger models.

**Review by FFroehlich — APPROVED** [review]
👍

File: python/amici/ode_export.py (hunk 3 above)
Comment (dweindl) [reply python/amici/ode_export.py:1888]: Thanks, makes sense

**Pull request merged**